inspire_etf_validator/domain/ngr.py

```diff
@@ -0,0 +1,163 @@
+    if not os.path.isfile(CACHE_FILENAME) or __cache_is_expired():
+        logger.info("downloading ngr record data...")
+        ngr_records = __get_all_ngr_records()
+        for ngr_record in ngr_records:
+            ngr_record.update(__get_record_info(ngr_record["uuid"]))
+
+        if enable_caching:
+            logger.info("writing all ngr record data to cache file " + CACHE_FILENAME)
+            with open(CACHE_FILENAME, "w", encoding="utf-8") as f:
+                json.dump(ngr_records, f, ensure_ascii=False, indent=4)
+    else:
+        logger.info("reading ngr records from cache file " + CACHE_FILENAME)
+        with open(CACHE_FILENAME) as infile:
+            ngr_records = json.load(infile)
+
+    return ngr_records
+
+
+def __cache_is_expired():
+    file_mod_time = datetime.fromtimestamp(
+        os.stat(CACHE_FILENAME).st_mtime
+    )  # This is a datetime.datetime object!
+    now = datetime.today()
+    max_delay = timedelta(seconds=CACHE_EXPIRATION_IN_SECONDS)
+    return now - file_mod_time > max_delay
+
+
+def __get_all_ngr_records():
+    ngr_records = []
+
+    start_position = "1"
+    while True:
+        records_base_url = (
+            URL_NGR
+            + "/srv/dut/csw-inspire?request=GetRecords&Service=CSW&Version=2.0.2&typeNames"
+            "=gmd:MD_Metadata&resultType=results&constraintLanguage=CQL_TEXT"
+            "&constraint_language_version=1.1.0&constraint=type='service'+AND"
+            "+organisationName='Beheer+PDOK'&startPosition=" + start_position
+        )
+        logger.info("fetching records_base_url: " + records_base_url)
+        response = requests.get(records_base_url, headers=REQUEST_HEADERS)
+        document = ET.fromstring(response.content)
+
+        ex_node = document.findall("./ows:ExceptionReport", NAMESPACE_PREFIXES)
+        if len(ex_node) > 0:
+            exception_code = document.find(
+                "./ows:ExceptionReport/ows:Exception/[@exceptionCode]",
+                NAMESPACE_PREFIXES,
+            ).text
+            if exception_code is not None:
+                raise Exception(
+                    "Exception in CSW response, exceptionCode: " + exception_code
+                )
+
+        start_position = document.find(
+            ".//csw:SearchResults/[@nextRecord]", NAMESPACE_PREFIXES
+        ).attrib["nextRecord"]
+
+        temp_records = document.findall(".//csw:SummaryRecord", NAMESPACE_PREFIXES)
+        for temp_record in temp_records:
+            title = temp_record.find("dc:title", NAMESPACE_PREFIXES).text
+            identifier = temp_record.find("dc:identifier", NAMESPACE_PREFIXES).text
+            label = title.replace(" ", "_").lower()
+
+            temp_record_result = {"uuid": identifier, "title": title, "label": label}
+            ngr_records.append(temp_record_result)
+
+        if start_position == "0":
+            break
+
+    return ngr_records
+
+
+def __get_service_type(service_access_point):
```

Comment:
Wat mij hier aan opvalt is dat dit kleine versus grote letters zijn. Is dat af te leiden?